packages/nimble-components/src/rich-text/models/tests/markdown-parser.spec.ts

```diff
@@ -0,0 +1,383 @@
+import { getSpecTypeByNamedList } from '../../../utilities/tests/parameterized';
+import { wackyStrings } from '../../../utilities/tests/wacky-strings';
+import { RichTextMarkdownParser } from '../markdown-parser';
+
+describe('Markdown parser', () => {
+    function getTagsFromDocumentFragment(doc: DocumentFragment): string[] {
```

Comment:
Can we delete the corresponding methods from the page objects now (e.g. `getRenderedMarkdownTagNames`) or are they still used somewhere?

Reply:
Yeah, that sounds good. 
But the page-objects used in editor or viewer component's spec obtains the data from the UI and the functions used in model's specs takes in the data as arguments to the functions and doesn't rely on UI.
Shall we add these functions in the page objects in addition to existing ones?

Reply:
I was imagining something like `models/testing/markdown-parser-utils.ts` which would contain `getTagsFromDocumentFragment` and `getLeafContentsFromDocumentFragment` with the same signature they have now. Then the methods in the element page object like `getRenderedMarkdownTagNames` would be updated to call those methods by passing in the relevant element / fragment.